fearless/Modules/Staking/SelectValidatorsFlow/ValidatorSearch/ValidatorSearchInteractor.swift

```diff
@@ -7,26 +7,34 @@ final class ValidatorSearchInteractor {
     let validatorOperationFactory: ValidatorOperationFactoryProtocol
     let operationManager: OperationManagerProtocol
 
+    private var currentOperation: CompoundOperationWrapper<[SelectedValidatorInfo]>?
+
     init(
         validatorOperationFactory: ValidatorOperationFactoryProtocol,
         operationManager: OperationManagerProtocol
     ) {
         self.validatorOperationFactory = validatorOperationFactory
         self.operationManager = operationManager
     }
+
+    private func cancelSearch() {
+        currentOperation?.cancel()
+        currentOperation = nil
+    }
 }
 
 extension ValidatorSearchInteractor: ValidatorSearchInteractorInputProtocol {
     func performValidatorSearch(accountId: AccountId) {
-        let operation = validatorOperationFactory
+        cancelSearch()
+
+        currentOperation = validatorOperationFactory
             .wannabeValidatorsOperation(for: [accountId])
 
-        operation.targetOperation.completionBlock = { [weak self] in
+        currentOperation!.targetOperation.completionBlock = { [weak self] in
             DispatchQueue.main.async {
                 do {
-                    let result = try operation.targetOperation.extractNoCancellableResultData()
-
-                    guard let validatorInfo = result.first else {
+                    guard let result = try self?.currentOperation?.targetOperation.extractNoCancellableResultData(),
```

Comment:
should we also nullify the currentOperation here?

Reply:
To my understanding it is not really necessary, since the operation itself is not going to produce any data at this point and for cancel function it doesn't really matter whether currentOperation is will or not. However, it might be worth to make code more precise. Let's do it

Reply:
There are several points why it is good to nullify:

1) Operation can contain logic that retains other objects, so it is expected that the operation is released when it is finished

2) Operation occupies memory

The first point is the most important I think.